internal/services/containers/container_group_resource.go

```diff
@@ -1477,16 +1478,56 @@ func expandContainerProbe(input interface{}) *containerinstance.ContainerProbe {
 
 				httpGetScheme := containerinstance.Scheme(scheme)
 				probe.HttpGet = &containerinstance.ContainerHttpGet{
-					Path:   pointer.FromString(path),
-					Port:   int64(port),
-					Scheme: &httpGetScheme,
+					Path:        pointer.FromString(path),
+					Port:        int64(port),
+					Scheme:      &httpGetScheme,
+					HttpHeaders: expandContainerProbeHttpHeaders(x["http_headers"].(map[string]interface{})),
 				}
 			}
 		}
 	}
 	return &probe
 }
 
+func expandContainerProbeHttpHeaders(input map[string]interface{}) *[]containerinstance.HttpHeader {
+	if len(input) == 0 {
+		return nil
+	}
+
+	headers := []containerinstance.HttpHeader{}
+	for k, v := range input {
+		header := containerinstance.HttpHeader{
+			Name:  pointer.FromString(k),
+			Value: pointer.FromString(v.(string)),
+		}
+		headers = append(headers, header)
+	}
+	sort.Slice(headers, func(i, j int) bool {
```

Comment:
how come we are sorting these? and could we leave a comment explaining it?

Reply:
The original itent there is to make the expand consistent on ordering. Whilst it actually makes less sense but more confusion. So I just removed this.